agrirouter-sdk-java-tests/src/test/java/com/dke/data/agrirouter/test/messaging/rest/SendChunkedMessageTest.java

```diff
@@ -0,0 +1,133 @@
+    final OnboardingResponse onboardingResponse =
+        OnboardingResponseRepository.read(OnboardingResponseRepository.Identifier.FARMING_SOFTWARE);
+
+    MessageHeaderParameters messageHeaderParameters = new MessageHeaderParameters();
+    messageHeaderParameters.setTechnicalMessageType(ContentMessageType.ISO_11783_TASKDATA_ZIP);
+    messageHeaderParameters.setApplicationMessageId(MessageIdService.generateMessageId());
+    messageHeaderParameters.setApplicationMessageSeqNo(
+        SequenceNumberService.generateSequenceNumberForEndpoint(onboardingResponse));
+    messageHeaderParameters.setMode(Request.RequestEnvelope.Mode.PUBLISH);
+
+    PayloadParameters payloadParameters = new PayloadParameters();
+    payloadParameters.setValue(fakeLargeMessageContent());
+    payloadParameters.setTypeUrl(SystemMessageType.EMPTY.getKey());
+
+    List<MessageParameterTuple> tuples =
+        encodeMessageService.chunk(messageHeaderParameters, payloadParameters, onboardingResponse);
+
+    List<String> encodedMessages = encodeMessageService.encode(tuples);
+
+    SendMessageParameters sendMessageParameters = new SendMessageParameters();
+    sendMessageParameters.setEncodedMessages(encodedMessages);
+    sendMessageParameters.setOnboardingResponse(onboardingResponse);
+    sendMessageService.send(sendMessageParameters);
+
+    waitForTheAgrirouterToProcessTheMessages();
+
+    FetchMessageService fetchMessageService = new FetchMessageServiceImpl();
+    Optional<List<FetchMessageResponse>> fetchMessageResponses =
+        fetchMessageService.fetch(
+            onboardingResponse,
+            new DefaultCancellationToken(MAX_TRIES_BEFORE_FAILURE, DEFAULT_INTERVAL));
+
+    Assertions.assertTrue(fetchMessageResponses.isPresent());
+    Assertions.assertEquals(EXPECTED_NUMBER_OF_CHUNKS, fetchMessageResponses.get().size());
+    Assertions.assertNotNull(fetchMessageResponses.get().get(0).getCommand());
+    Assertions.assertNotNull(fetchMessageResponses.get().get(1).getCommand());
+    Assertions.assertNotNull(fetchMessageResponses.get().get(2).getCommand());
+
+    Message firstAck = fetchMessageResponses.get().get(0).getCommand();
+    Message secondAck = fetchMessageResponses.get().get(1).getCommand();
+    Message thirdAck = fetchMessageResponses.get().get(2).getCommand();
+
+    Arrays.stream(new Message[] {firstAck, secondAck, thirdAck})
+        .forEach(
+            message -> {
+              DecodeMessageService decodeMessageService = new DecodeMessageServiceImpl();
+              DecodeMessageResponse decodeMessageResponse =
+                  decodeMessageService.decode(message.getMessage());
+
+              Assertions.assertMatchesAny(
+                  Arrays.asList(HttpStatus.SC_OK, HttpStatus.SC_CREATED, HttpStatus.SC_NO_CONTENT),
+                  decodeMessageResponse.getResponseEnvelope().getResponseCode());
+            });
+  }
+
+  @Test
+  void determineMaxChunkLengthForNonEncodedChunks() {
+    String base64EncodedString =
```

Comment:
This test provides no knowledge about the SDK at all. If it is used to check the threshold 767997 for the developer, it might make sense to create a constant from that value. Otherwise there's no connection to the other tests